### ISSUE TITLE
[TODO] re-implement limit_train/dev/test flags

### ISSUE THREAD

**Issue by rhamnett** [mozilla/DeepSpeech#2777]
Add code to ensure the flags below work.
 
f.DEFINE_integer('limit_train', 0, 'maximum number of elements to use from train set - 0 means no limit')
f.DEFINE_integer('limit_dev', 0, 'maximum number of elements to use from validation set- 0 means no limit')
f.DEFINE_integer('limit_test', 0, 'maximum number of elements to use from test set- 0 means no limit')

**Comment by lissyx** [comment]
> Add code to ensure the flags below work.
> 
> f.DEFINE_integer('limit_train', 0, 'maximum number of elements to use from train set - 0 means no limit')
> f.DEFINE_integer('limit_dev', 0, 'maximum number of elements to use from validation set- 0 means no limit')
> f.DEFINE_integer('limit_test', 0, 'maximum number of elements to use from test set- 0 means no limit')

Can you elaborate? Do you have STR to reproduce a bug ?

**Comment by lissyx** [comment]
Hm this was removed as part of 1cea2b0fe88b888ae8bbbb4cbe2743c1a6087552 last year, was it a mistake or on purpose? I don't remember cc @reuben

**Comment by rhamnett** [comment]
It's literally not in the code base. It's up to you. It's easy to create a new, limited CSV file but it seems better just have a flag so I was prepared to re-implement it.

**Comment by reuben** [comment]
It was an oversight on my part when porting the feeding code to tf.data and then I never got around to fixing it. Should be simple enough to add back the limits.

**Comment by lissyx** [comment]
@rhamnett Should be a fairly easy PR, do you want to try ?

**Comment by tilmankamp** [comment]
I'd put it on hold till #2723 lands, as it'd have to be re-implemented completely.

**Comment by rhamnett** [comment]
I will sort it yeah but I'll take @tilmankamp's advice

**Comment by lissyx** [comment]
I'm wondering since when this is broken. Checking `v0.4.1`, there's `limit` as an argument to `DataSet`, but it's never used anywhere.

**Comment by lissyx** [comment]
Ok, actual removal of the feature seems to have been in 44e502e236d676dfcdb3068f6a6d9d1a9d644dd1

**Comment by tilmankamp** [comment]
How about something like
`--train_files some/data/set.csv[10:-100],some/other/data.sdb[:100]` ?
Should be straight-forward to implement as extended generator functions in `util.sample_collections.SDB` and `util.sample_collections.CSV`.

**Comment by lissyx** [comment]
> How about something like
> `--train_files some/data/set.csv[10:-100],some/other/data.sdb[:100]` ?
> Should be straight-forward to implement through extended generator functions in `util.sample_collections.SDB` and `util.sample_collections.CSV`.

I was looking into `create_dataset` and re-vive the `--limit` flags. I worry that the proposed syntax might be unobvious to people and error prone from shell point of view

**Comment by lissyx** [comment]
> Should be straight-forward to implement through extended generator functions in `util.sample_collections.SDB` and `util.sample_collections.CSV`.

Also, wouldn't that limit the use to SDB only ?

**Comment by lissyx** [comment]
> > Should be straight-forward to implement through extended generator functions in `util.sample_collections.SDB` and `util.sample_collections.CSV`.
> 
> Also, wouldn't that limit the use to SDB only ?

Nevermind, I'm reading the source, my mind was lost somewhere else.

**Comment by reuben** [comment]
Closing in favor of #1565.
